pkg/auth/handler/sso/auth_handler_test.go

```diff
@@ -151,62 +166,35 @@ func TestAuthHandler(t *testing.T) {
 			So(resp.Code, ShouldEqual, 302)
 			So(resp.Header().Get("Location"), ShouldEqual, "http://localhost:3000")
 
-			// check cookies
-			// it should have following format
-			// {
-			// 	"callback_url": "callback_url"
-			// 	"result": authResp
-			// }
-			cookies := resp.Result().Cookies()
-			So(cookies, ShouldNotBeEmpty)
-			var ssoDataCookie *http.Cookie
-			for _, c := range cookies {
-				if c.Name == "sso_data" {
-					ssoDataCookie = c
-					break
-				}
-			}
-			So(ssoDataCookie, ShouldNotBeNil)
-
-			// decoded it first
-			decoded, err := base64.StdEncoding.DecodeString(ssoDataCookie.Value)
-			So(err, ShouldBeNil)
-			So(decoded, ShouldNotBeNil)
-
-			// Unmarshal to map
-			data := make(map[string]interface{})
-			err = json.Unmarshal(decoded, &data)
-			So(err, ShouldBeNil)
-
-			// check callback_url
-			So(data["callback_url"], ShouldEqual, "http://localhost:3000")
-
-			// check result(authResp)
-			authResp, err := json.Marshal(data["result"])
+			actual, err := decodeCookie(resp)
 			So(err, ShouldBeNil)
 			p, err := sh.OAuthAuthProvider.GetPrincipalByProviderUserID(providerName, providerUserID)
 			So(err, ShouldBeNil)
 			token := mockTokenStore.GetTokensByAuthInfoID(p.UserID)[0]
-			So(authResp, ShouldEqualJSON, fmt.Sprintf(`{
+			So(actual, ShouldEqualJSON, fmt.Sprintf(`
+			{
+				"callback_url": "http://localhost:3000",
 				"result": {
-					"user": {
-						"id": "%s",
-						"is_verified": false,
-						"is_disabled": false,
-						"last_login_at": "2006-01-02T15:04:05Z",
-						"created_at": "0001-01-01T00:00:00Z",
-						"verify_info": {},
-						"metadata": {}
-					},
-					"identity": {
-						"id": "%s",
-						"type": "oauth",
-						"provider_id": "mock",
-						"provider_user_id": "mock_user_id",
-						"raw_profile": {},
-						"claims": {}
-					},
-					"access_token": "%s"
+					"result": {
```

Comment:
Is the double nested results intentional?

Reply:
Yes. The first level is `callback` and `result`. This `result` denotes the http response body. The second level is either `result` or `error` which is our conventional API response shape.